golos.ctrl/src/golos.ctrl.cpp

```diff
@@ -351,6 +363,19 @@ void control::send_witness_event(const witness_info& wi) {
     eosio::event(_self, "witnessstate"_n, std::make_tuple(wi.name, wi.total_weight)).send();
 }
 
+void control::active_witness(name witness, bool flag) {
+    // TODO: simplify upsert to allow passing just inner lambda
+    bool exists = upsert_tbl<witness_tbl>(witness, [&](bool) {
+        return [&](witness_info& w) {
+            eosio_assert(w.active, "witness already stopped");
```

Comment:
This assert invalid. What will happen in case when witness inactive and flag==1?